pkg/cloud/aws/actuators/machine/actuator_test.go

```diff
@@ -34,6 +34,153 @@ func init() {
+			machine: machine,
+			error:   awsServiceError,
+			operation: func(actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine) {
+				actuator.CreateMachine(cluster, machine)
+			},
+			event: "Warning FailedCreate CreateError",
+		},
+		{
+			name:    "Create machine event failed (error launching instance)",
+			machine: machine,
+			error:   launcInstanceError,
+			operation: func(actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine) {
+				actuator.CreateMachine(cluster, machine)
+			},
+			event: "Warning FailedCreate CreateError",
+		},
+		{
+			name:    "Create machine event failed (error updating load balancers)",
+			machine: machine,
+			error:   lbError,
+			operation: func(actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine) {
+				actuator.CreateMachine(cluster, machine)
+			},
+			event: "Warning FailedCreate CreateError",
+		},
+		{
+			name:    "Create machine event succeed",
+			machine: machine,
+			operation: func(actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine) {
+				actuator.CreateMachine(cluster, machine)
+			},
+			event: "Normal Created Created Machine aws-actuator-testing-machine",
+		},
+		{
+			name:    "Delete machine event failed",
+			machine: machineInvalidProviderConfig,
+			operation: func(actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine) {
+				actuator.DeleteMachine(cluster, machine)
+			},
+			event: "Warning FailedDelete InvalidConfiguration",
+		},
+		{
+			name:    "Delete machine event succeed",
+			machine: machine,
+			operation: func(actuator *Actuator, cluster *clusterv1.Cluster, machine *clusterv1.Machine) {
+				actuator.DeleteMachine(cluster, machine)
+			},
+			event: "Normal Deleted Deleted Machine aws-actuator-testing-machine",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			mockCtrl := gomock.NewController(t)
+			mockAWSClient := mockaws.NewMockClient(mockCtrl)
+
+			eventsChannel := make(chan string, 1)
+
+			params := ActuatorParams{
+				Client:     fake.NewFakeClient(tc.machine),
+				KubeClient: kubernetesfake.NewSimpleClientset(awsCredentialsSecret, userDataSecret),
+				AwsClientBuilder: func(kubeClient kubernetes.Interface, secretName, namespace, region string) (awsclient.Client, error) {
+					if tc.error == awsServiceError {
+						return nil, fmt.Errorf(awsServiceError)
+					}
+					return mockAWSClient, nil
+				},
+				Codec: codec,
+				// use fake recorder and store an event into one item long buffer for subsequent check
+				EventRecorder: &record.FakeRecorder{
+					Events: eventsChannel,
+				},
+			}
+
+			if tc.error == launcInstanceError {
+				mockRunInstances(mockAWSClient, true)
+			} else {
+				mockRunInstances(mockAWSClient, false)
+			}
+
+			mockDescribeInstances(mockAWSClient, false)
+			mockTerminateInstances(mockAWSClient)
+
+			if tc.error == lbError {
```

Comment:
Could be reduced to:

    mockRegisterInstancesWithLoadBalancer(mockAWSClient, <bool-expression>)

Reply:
Done